pyproject.toml

```diff
@@ -1,47 +1,203 @@
 [build-system]
-requires = ["setuptools>=40.8.0", "wheel", "cython"]
-build-backend = "setuptools.build_meta:__legacy__"
+requires = ["setuptools>=64", "setuptools_scm>=8", "wheel"]
+build-backend = "setuptools.build_meta"
 
+[project]
+authors = [{name = "The HyperSpy Community"}]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: GNU General Public License v3 (GPLv3)",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Topic :: Software Development :: Libraries",
+]
+dependencies = [
+    "cloudpickle",
+    "dask[array]>=2021.3.1",
+    "importlib-metadata>=3.6",
+    "jinja2",
+    "matplotlib>=3.1.3",
+    "natsort",
+    "numpy>=1.20.0",
+    "packaging",
+    "pint>=0.10",
+    "pooch",
+    "prettytable>=2.3",
+    "python-dateutil>=2.5.0",
+    "pyyaml",
+    "requests",
+    "rosettasciio[hdf5]",
+    "scikit-image>=0.18",
+    "scipy>=1.5.0",
+    "sympy>=1.6",
+    "tqdm>=4.9.0",
+    "traits>=4.5.0",
+]
+description = "Multidimensional data analysis toolbox"
+dynamic = ["version"]
+keywords = [
+    "EDX",
+    "EELS",
+    "EFTEM",
+    "EMSA",
+    "FEI",
+    "ICA",
+    "PCA",
+    "PES",
+    "STEM",
+    "TEM",
+    "curve fitting",
+    "data analysis",
+    "dm3",
+    "electron energy loss spectroscopy",
+    "electron microscopy",
+    "emi",
+    "energy dispersive x-rays",
+    "hyperspectral",
+    "hyperspectrum",
+    "hyperspy",
+    "machine learning",
+    "microscopy",
+    "model",
+    "msa",
+    "multidimensional",
+    "numpy",
+    "python",
+    "quantification",
+    "scipy",
+    "ser",
+    "spectroscopy",
+    "spectrum image",
+]
+name = "hyperspy"
+readme = "README.rst"
+requires-python = ">=3.8"
+
+[project.license]
+file = "COPYING.txt"
+
+[project.optional-dependencies]
+all = [
+    "hyperspy[gui-jupyter]",
+    "hyperspy[gui-traitsui]",
+    "hyperspy[ipython]",
+    "hyperspy[learning]",
+    "hyperspy[speed]",
+]
+coverage = [
+    "pytest-cov",
+]
+dev = [
+    "black",
+    "hyperspy[all]",
+    "hyperspy[coverage]",
+    "hyperspy[doc]",
+    "hyperspy[tests]",
+]
+doc = [
+    "pydata_sphinx_theme",
```

Comment:
Already add ``"numpydoc"`` for compatibility with #3273?

Reply:
They will merge conflict anyway, so I prefer to leave it to #3273.